### ISSUE TITLE
bug: `stencil build` command generates the folder `dist\esm\polyfills\` folder even if buildEs5=false in config

### ISSUE THREAD

**Issue by alvarocjunq** [ionic-team/stencil#5416]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.12.4

### Current Behavior

Even if set the flag `buildEs5: false` in `stencil.config.ts` the output in dist is generating the es5 files:

<img width="529" alt="image" src="https://github.com/ionic-team/stencil/assets/159884426/9435a275-411e-4e8b-a967-951d1cbb1c29">

In my company they are finding vulnerabilities in this files that we don't even use 😢 

### Expected Behavior

It is possible not to generate these files that are not being used by the component?

### System Info

```shell
System: node 18.16.0
    Platform: windows (10.0.22621)
   CPU Model: 13th Gen Intel(R) Core(TM) i5-1345U (12 cpus)
    Compiler: C:\Projects\web-components\axa.be.webcomponent.abe-dropdown\node_modules\@stencil\core\compiler\stencil.js
       Build: 1707755243
     Stencil: 4.12.2
  TypeScript: 5.3.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.27.0
```

### Steps to Reproduce

1) Generate a new Stencil Component
2) Configure the `stencil.config.ts` with the flag `buildEs5: false`
3) Execute `npm run build`

### Code Reproduction URL

https://github.com/alvarocjunq/stencil-es5-polyfill-issue

**Comment by alvaro-junqueira-external-axa** [comment]
Thank you to take into account @rwaskiewicz 😄

**Comment by alvarocjunq** [comment]
Thank you to take into account @rwaskiewicz 😄

**Comment by alicewriteswrongs** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.18.1 release](https://github.com/ionic-team/stencil/releases/tag/v4.18.1)!
